packages/sdk/src/machines/ram-machine/index.ts

```diff
@@ -303,17 +305,38 @@ export const ramMachine = setup({
           hash: txHash,
         });
 
-        // If the transaction was reverted, resimulate it to get the revert reason
         if (txReceipt.status === "reverted") {
-          // These if statements are necessary for type checking
-          if (txRequest.functionName === "createBid") {
-            // Resimulate the transaction to get the revert reason
-            await publicClient.simulateContract(txRequest);
-          }
+          // If the transaction was reverted attempt to use the public client
+          // to call trace_transaction to get the revert reason. This is not
+          // universally supported by all JSON RPC providers so we must wrap
+          // this call in a try/catch block. Fallback to resimulating the tx
+          // if the trace_transaction call fails. Because chain state may have
+          // been updated between the receipt being returned and the trace
+          // being fetched it is possible that the revert reason will not be
+          // accurate in this case.
+          try {
+            const traceResult = await traceTransaction(publicClient, txHash);
+
+            throw new ContractFunctionRevertedError({
+              abi: minterRAMV0Abi,
+              functionName: txRequest.functionName,
+              data: traceResult?.[0]?.result?.output,
```

Comment:
Just to check, does this get properly decoded on the frontend? I think the revert reason is in args[0], but maybe we are accessing that when decoding later on?